### ISSUE TITLE
Starting teleport by script

### ISSUE THREAD

**Issue by nixtwiz** [thestonefox/VRTK#685]
In my game the teleport charges up when holding the grip button and will teleport when you release the grip button. How do I start the teleport on grip release?

**Comment by nixtwiz** [comment]
Also is this the right place to ask this?

**Comment by thestonefox** [comment]
To enable the teleporter to work it just needs to receive a destination marker event.

How are you determining where the destination for the teleport should be? E.g. pointing a pointer at the ground where you want to teleport to.

**Comment by nixtwiz** [comment]
I'm using a bezier pointer. Also in case it makes a difference, I'm using dash instead of teleport.

**Comment by thestonefox** [comment]
The way the pointers set the destination is based on a `VRTK_ControllerEvents` parameter called `pointerSetButton` but at the moment the button aliases all work on button down.

I'll look to add in an option so you can set an alias as 'button up' as well then you'll just be able to change the `pointerSetButton` to `grip up`

**Comment by nixtwiz** [comment]
Awesome thanks!

**Comment by thestonefox** [comment]
Actually looking at the code the pointer set event is thrown when the button is released anyway.

So if you just change the `Pointer Set Alias` in the `VRTK_ControllerEvents` script that is on the controller to `Grip` then you will teleport when you release the grip button.

However, this may not achieve what you're hoping for because the Pointer Set Event will be emitted straight away when you release. so a simple click of the grip will initiate a teleport.

**Comment by thestonefox** [comment]
Looking at the code again, having "on button up" for pointer aliases doesn't make sense either.

The pointer alias is checking the state of a button for whether it's being pressed or not, so having "grab" on button up wouldn't make much sense.

I think i'm going to revert the title of this ticket and not implement the button up for aliases.

**Comment by nixtwiz** [comment]
Sorry for your time, the problem was on my end.
My problem now is how do I only activate the pointer at certain times and then deactivate it after teleporting? Right now I've just set the beamActive bool to public and set that to true after the grip has been held long enough, then false after it has been let go. Doing so causes the beam to stay there after the teleport until I teleport again, so essentially I can walk around and look at the beam from when I teleported. Is there a better way to do this?
